lib/PrintMiniRust.ml

```diff
@@ -236,6 +236,9 @@ and print_expression_with_block env (e: expr): document =
 
 and print_statements env (e: expr): document =
   match e with
+  | Let ({ typ = Unit; _ }, Unit, e2) ->
```

Comment:
do you understand why this is not caught by the previous passes over the `Ast`?

Reply:
Some of them are introduced by the translation of EBufFree in the Ast to MiniRust pass, which thus occurs after the Ast simplification passes. I'm not sure this is the case for all, but that at least gives some unhandled cases.

Reply:
cool, if you can, a comment would help -- so many other phases *already* do this earlier that it's a little surprising to see this here; it makes perfect sense! but a comment would help a bit

Reply:
Good point, this is done in [a62fa60](https://github.com/FStarLang/karamel/pull/463/commits/a62fa609c9309d4be2c2eb9a44c32446245fbc6a)